### ISSUE TITLE
False positives in `length-zero-no-unit` with new math functions

### ISSUE THREAD

**Issue by romainmenke** [stylelint/stylelint#6870]
### What steps are needed to reproduce the bug?

```css
.a {
  width: atan2(0px, 1px);
}
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "length-zero-no-unit": true
  }
}
```

### How did you run Stylelint?

https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0BDABMAOgOzWg7gJZQAuAFvGiiStgEwAUADAA4AeANGgIzsCUAbhwBfEBxAAzQgBsYAORQBbOIhhslLWUjABnHWPARsUgOYIQWXGkwgATgFdZOm5Ut48N2dhPkAtAC8YWwhfbBD7bEISFzQAbRIHGABdHDxhEQNIY0ITADEIW0VqcwArHSMDWBZ9RDdrEB0SAE9ZaUJsaIR66WoYRpsOVPrGlpg2jt8s019G2igUWygYz17+kAzhIA

### Which version of Stylelint or dependencies are you using?

latest

### What did you expect to happen?

No error on any of the new math functions

### What actually happened?

```
Unexpected unit (length-zero-no-unit)[2:17-19]
```

### Proposal to fix the bug

Add the new math function to `/lib/reference`
